core/src/main/java/org/apache/calcite/sql/dialect/SnowflakeSqlDialect.java

```diff
@@ -288,6 +284,34 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
       DateTimestampFormatUtil dateTimestampFormatUtil = new DateTimestampFormatUtil();
       dateTimestampFormatUtil.unparseCall(writer, call, leftPrec, rightPrec);
       break;
+    case "LOWER":
+      if (call.operand(0).getKind() == SqlKind.LITERAL) {
+        getDayofWeek(writer, call, leftPrec, rightPrec);
+      } else {
+        super.unparseCall(writer, call, leftPrec, rightPrec);
+      }
+      break;
+    default:
+      super.unparseCall(writer, call, leftPrec, rightPrec);
+    }
+  }
+
+  private void getDayofWeek(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    switch (((SqlLiteral) call.operand(0)).getValueAs(String.class)) {
+    case "Sunday":
+    case "Monday":
+    case "Tuesday":
+    case "Wednesday":
+    case "Thursday":
+    case "Friday":
+    case "Saturday":
+      final SqlWriter.Frame operatorFrame = writer.startFunCall(call.getOperator().toString());
+      final SqlWriter.Frame substrFrame = writer.startFunCall("SUBSTR");
+      writer.print(call.operand(0).toString());
```

Comment:
replace it with `call.operand(0).unparse(writer, call, leftPrec, rightPrec)`

Reply:
done